src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1975,6 +1980,17 @@ private void VisitSelect(Expression operand)
             }
         }
 
+        private string RemoveTransperentIdentifiersIfNeeded(MemberExpression memberExpression)
+        {
+            var selectPath = GetSelectPath(memberExpression);
+            while (selectPath.StartsWith("<>h__TransparentIdentifier"))
+            {
+                selectPath = selectPath.Substring(28);
```

Comment:
Are you sure this is `28` ? `<>h__TransparentIdentifier` is just 26 chars?

Reply:
yes, 26 + 1 for the number suffix and +1 for "."
i.e :  **<>h__TransparentIdentifier0.**Name